smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/tracing/TracingPropagationTest.java

```diff
@@ -0,0 +1,390 @@
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.header.Header;
+import org.apache.kafka.common.header.internals.RecordHeaders;
+import org.apache.kafka.common.serialization.IntegerDeserializer;
+import org.apache.kafka.common.serialization.IntegerSerializer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.messaging.Outgoing;
+import org.jboss.weld.environment.se.Weld;
+import org.jboss.weld.environment.se.WeldContainer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.grpc.Context;
+import io.opentelemetry.OpenTelemetry;
+import io.opentelemetry.exporters.inmemory.InMemorySpanExporter;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.export.SimpleSpanProcessor;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.SpanContext;
+import io.opentelemetry.trace.SpanId;
+import io.opentelemetry.trace.TraceId;
+import io.opentelemetry.trace.TracingContextUtils;
+import io.reactivex.Flowable;
+import io.smallrye.config.SmallRyeConfigProviderResolver;
+import io.smallrye.reactive.messaging.TracingMetadata;
+import io.smallrye.reactive.messaging.kafka.KafkaConnector;
+import io.smallrye.reactive.messaging.kafka.KafkaTestBase;
+import io.smallrye.reactive.messaging.kafka.KafkaUsage;
+import io.smallrye.reactive.messaging.kafka.MapBasedConfig;
+
+public class TracingPropagationTest extends KafkaTestBase {
+
+    private WeldContainer container;
+    private InMemorySpanExporter testExporter;
+    private SpanProcessor spanProcessor;
+
+    @Before
+    public void setup() {
+        testExporter = InMemorySpanExporter.create();
+        spanProcessor = SimpleSpanProcessor.newBuilder(testExporter).build();
+        OpenTelemetrySdk.getTracerProvider().addSpanProcessor(spanProcessor);
+    }
+
+    @After
+    public void cleanup() {
+        if (testExporter != null) {
+            testExporter.shutdown();
+        }
+        if (spanProcessor != null) {
+            spanProcessor.shutdown();
+        }
+
+        if (container != null) {
+            container.close();
+        }
+
+        // Release the config objects
+        SmallRyeConfigProviderResolver.instance().releaseConfig(ConfigProvider.getConfig());
+    }
+
+    @Test
+    public void testFromAppToKafka() {
```

Comment:
Help me understand what this test does. 

The test class sends messages from producer to consumer. The span context is injected and extracted by the test class so what does it test? I am probably mising something here.